File: C++/70. Climbing Stairs.cpp
class Solution
{
public:
    int climbStairs(int n)
    {
        if (n <= 1)
            return 1;


        int prev = 1, current = 1, next;

        for (int i = 2; i <= n; ++i)
        {
            next = prev + current;
            prev = current;
            current = next;
        }
        return current;
    }
};

File: Python/70. Climbing Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 1:
            return 1

        prev, current = 1, 1

        for i in range(2, n + 1):
            next_val = prev + current
            prev, current = current, next_val

        return current